### PR TITLE
Block Bindings: Update source registration syntax and remove APIs that should be private

### PR DESCRIPTION
**Pull request by michalczaplinski**
There are a few changes in this PR: 

Following https://github.com/WordPress/wordpress-develop/pull/5888 I've:
- Moved the source registration to an "array" syntax
- Removed the `replace_html()` method of the `WP_Block_Bindings` class
- Moved the body of `replace_html()` from the `WP_Block_Bindings` class to the `block-bindings.php` file

### CODE REVIEW

File: lib/compat/wordpress-6.5/block-bindings/block-bindings.php [hunk 1]
```diff
@@ -62,8 +68,98 @@ function wp_block_bindings_get_sources() {
  * @param string $source_value The value used to replace the HTML.
  * @return string The modified block content.
  */
-if ( ! function_exists( 'wp_block_bindings_replace_html' ) ) {
-	function wp_block_bindings_replace_html( $block_content, $block_name, $block_attr, $source_value ) {
-		return wp_block_bindings()->replace_html( $block_content, $block_name, $block_attr, $source_value );
+if ( ! function_exists( 'gutenberg_block_bindings_replace_html' ) ) {
```
Comment (gziolo) [inline lib/compat/wordpress-6.5/block-bindings/block-bindings.php]: Nit: there is no need to guard Gutenberg functions for redeclaration. It’s only safe guard required for functions and classes expected to land in WP core.

File: lib/compat/wordpress-6.5/block-bindings/sources/pattern.php [hunk 2]
```diff
@@ -29,7 +29,9 @@
 	};
 	wp_block_bindings_register_source(
 		'pattern_attributes',
-		__( 'Pattern Attributes', 'gutenberg' ),
-		$pattern_source_callback
+		array(
```
Comment (gziolo) [inline lib/compat/wordpress-6.5/block-bindings/sources/pattern.php:32]: There should be no textdomain used for translations expected to land in core.

File: lib/compat/wordpress-6.5/block-bindings/block-bindings.php (hunk 1 above)
Comment (michalczaplinski) [reply lib/compat/wordpress-6.5/block-bindings/block-bindings.php]: gotcha, thanks! Fixed in de089f650d9c1b00513818f2e1a17afdbeeca958

File: lib/compat/wordpress-6.5/block-bindings/sources/pattern.php (hunk 2 above)
Comment (michalczaplinski) [reply lib/compat/wordpress-6.5/block-bindings/sources/pattern.php:32]: Thanks! I removed it for both the `pattern` and `post-meta` sources in c9a245e0be4dde85535535b84441c9a6914c00b9

**Review by artemiomorales — APPROVED** [review]
Pending passing tests, looks good to me!

**Review by SantosGuillamot — APPROVED** [review]
I've tested it, and everything seems to keep working fine 🙂

File: lib/compat/wordpress-6.5/block-bindings/block-bindings.php [hunk 3]
```diff
@@ -62,8 +68,96 @@ function wp_block_bindings_get_sources() {
...
+			// TODO: Support for CSS selectors whenever they are ready in the HTML API.
+			// In the meantime, support comma-separated selectors by exploding them into an array.
+			$selectors = explode( ',', $block_type->attributes[ $block_attr ]['selector'] );
+			// Add a bookmark to the first tag to be able to iterate over the selectors.
+			$block_reader->next_tag();
+			$block_reader->set_bookmark( 'iterate-selectors' );
+
+			// TODO: This shouldn't be needed when the `set_inner_html` function is ready.
+			// Store the parent tag and its attributes to be able to restore them later in the button.
+			// The button block has a wrapper while the paragraph and heading blocks don't.
```
Comment (cbravobernal) [inline lib/compat/wordpress-6.5/block-bindings/block-bindings.php:92]: If you are backporting this file, make sure that multilines comments are with /* */

Comment (michalczaplinski) [reply lib/compat/wordpress-6.5/block-bindings/block-bindings.php:92]: Thanks Carlos! The backport is here btw https://github.com/WordPress/wordpress-develop/pull/5888 if you ever want to have a look.

**Comment by getdave** [comment]
I noticed this PR was merged after I raised the [PHP Sync Tracking Issue for WP 6.5](https://github.com/WordPress/gutenberg/issues/57959) and has changed PHP files that _may_ need backporting to WP Core.

Please forgive the ping, but I marked as `Needs PHP backport` and also added to the tracking Issue.

**Comment by michalczaplinski** [comment]
@getdave Backported to Core in https://github.com/WordPress/wordpress-develop/pull/5888

**Pull request merged**